src/math2d/point.rs

```diff
@@ -0,0 +1,23 @@
+use std::ops::Add;
+
+use super::vector::Vector;
+
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd)]
+pub struct Point {
+    pub x: f32,
+    pub y: f32,
```

Comment:
I implemented it with public members like you have in your C++, but to be consistent with the other data structures, maybe these should be private as well, and there could be a public get_coordinates method, which can return a tuple (x,y)?

Reply:
Making them private is prolly better. Accessing the values via get_coordinates() seems like a good idea